webserver/containers.go

```diff
@@ -4,111 +4,138 @@ import (
+		return errors.Wrapf(err, "fail to encode container usage payload")
+	}
 	return nil
 }
 
-func (c Controller) ContainerMemUsageHandler(res http.ResponseWriter, req *http.Request, params map[string]string) error {
-	id := mux.Vars(req)["id"]
+func (c Controller) ContainerMemUsageHandler(res http.ResponseWriter, _ *http.Request, params map[string]string) error {
+	id := params["id"]
 
 	containerMemoryUsage, err := c.mem.GetMemoryUsage(id)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "fail to get container memory usage")
 	}
 
 	res.WriteHeader(200)
-	json.NewEncoder(res).Encode(&containerMemoryUsage)
+	err = json.NewEncoder(res).Encode(&containerMemoryUsage)
+	if err != nil {
+		return errors.Wrapf(err, "fail to encode container memory usage payload")
+	}
 	return nil
 }
 
-func (c Controller) ContainerCpuUsageHandler(res http.ResponseWriter, req *http.Request, params map[string]string) error {
+func (c Controller) ContainerCPUUsageHandler(res http.ResponseWriter, _ *http.Request, params map[string]string) error {
 	id := params["id"]
 
 	containerCpuUsage, err := c.cpu.GetContainerUsage(id)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "fail to get container cpu usage")
 	}
+
 	res.WriteHeader(200)
-	json.NewEncoder(res).Encode(&containerCpuUsage)
+	err = json.NewEncoder(res).Encode(&containerCpuUsage)
+	if err != nil {
+		return errors.Wrapf(err, "fail to encode container cpu usage payload")
+	}
 	return nil
 }
 
 func (c Controller) ContainerNetUsageHandler(res http.ResponseWriter, req *http.Request, params map[string]string) error {
 	id := params["id"]
+
 	containerNet, err := c.net.GetUsage(id)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "fail to get container network usage")
 	}
 
 	res.WriteHeader(200)
-	json.NewEncoder(res).Encode(&containerNet)
+	err = json.NewEncoder(res).Encode(&containerNet)
+	if err != nil {
+		return errors.Wrapf(err, "fail to encode container network usage payload")
+	}
 	return nil
 }
 
 func (c Controller) ContainersUsageHandler(res http.ResponseWriter, req *http.Request, params map[string]string) error {
 	log := logger.Get(req.Context())
+
 	usage := client.NewContainersUsage()
 	containers, err := docker.ListContainers()
 	if err != nil {
 		res.WriteHeader(500)
 		log.WithError(err).Error("Fail to list containers")
-		errors := map[string]string{"message": "fail to list containers", "error": err.Error()}
-		json.NewEncoder(res).Encode(&errors)
+
+		errs := map[string]string{"message": "fail to list containers", "error": err.Error()}
+		err := json.NewEncoder(res).Encode(&errs)
+		if err != nil {
+			return errors.Wrapf(err, "fail to encode containers usage error payload")
+		}
 		return nil
 	}
+
 	for _, container := range containers {
 		cpuUsage, err := c.cpu.GetContainerUsage(container.ID)
 		if err != nil {
-			log.WithError(err).Errorf("Fail to get CPU usage of '%v'", container.ID)
+			log.WithError(err).Warnf("Fail to get CPU usage of '%v'", container.ID)
```

Comment:
We do'nt use the warn level in our Go projects: https://www.notion.so/scalingooriginal/Logs-in-Go-cb165968df1147fd93eb46df781a3eda#dc6c93608a0742c0a53e48201404ec9c